TrainingLoop/Callbacks/CSVLogger.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import ModelSupport
+
+/// A callback-based handler for logging the statistics to CSV file.
+public class CSVLogger {
+  public var path: String
+
+  let foundationFS: FoundationFileSystem
+  let foundationFile: FoundationFile
+
+  /// Create an instance that log statistics during the training loop.
+  public init(withPath path: String = "run/log.csv") {
+    self.path = path
+    self.foundationFS = FoundationFileSystem()
+    self.foundationFile = FoundationFile(path: path)
+  }
+
+  /// The callback used to hook into the TrainingLoop for logging statistics.
```

Comment:
This is a function, so you say what it *does*, per 

> Describe what a function or method does and what it returns, omitting null effects and Void returns

Please read https://swift.org/documentation/api-design-guidelines/#fundamentals

```suggestion
  /// Logs the end of an iteration of `loop`; ignoring other events.
```

or something.

Also, please describe the conditions under which it might throw.  It is probably enough to say that filesystem errors are reported by throwing.

Reply:
Done.